### ISSUE TITLE
UWP support

### ISSUE THREAD

**Issue by coonmoo** [praeclarum/NGraphics#44]
Would it be possible to add UWP support?

It seems that SharpDX 3 supports UWP.

Cheers

**Comment by jacobrillema** [comment]
Any update on this one?

**Comment by alllohaaa** [comment]
At least give us hope that this is in progress, please

**Comment by BlueRaja** [comment]
In case anyone doesn't know what UWP is: https://en.wikipedia.org/wiki/Universal_Windows_Platform
It's the new Windows API, used by Windows 10 and Windows 10 Phone apps.

**Comment by chrfalch** [comment]
Sorry, been really busy for the past few weeks.

**Comment by charlenni** [comment]
Thank you for your work!

I would like to see this too.

**Comment by damian-666** [comment]
Because Microsoft bought Xamarin and its now free, i'm giving this another
look as well. C# just got a big boost with this.   For MSFT I would use the
universal app approach using XAML for  the UI , i think, in  a grid,
using the borders for the UI,  with the content inside.     Not sure on
this.
Not sure what i will use for a scene graph ( retained mode graphics) ,
still, but the one in Canvas is so bad i manually hide and show stuff that
is off-screen every frame.    I wish i were in a position to donate but i'm
already pushing my limits.

**Comment by KSemenenko** [comment]
It is very cool issue, any update on this one?

**Comment by damian-666** [comment]
this would be integration of open scene graph or similar oct tree a spatial
index... someone could make a pull...   Inkscape itself has a terrible
 (nonexistent ) scene graph and some many bug and regressions that is is
not that easy to keep.  I was converting to XMAL and re importing but I
would love to work directly with SVG, into Xamarin, or native WPF,
Silverlight, or whatever MSFT is doign with c#.. ( there are enough users
screaming  c++ and COM  were not fun 15 years ago and not fun today.)
just to build some stuff.. i spend over 45 minuts!!!    matching the
msvcrt.. the GLUT..   blah  GLEW... blahh...  memories from decades are
coming back.. unwelcome .  hard times....  with the gpu... i do not think
c++ has anything over c# and I will be proving it to myself all week.  I
think i can port code faster than i can build the c++ crap.

.   i just want a canvas.  i want WPF , SL or similar  and high performance
in retained mode.   Apple has retained mode.   Android does not.    I do
deep zooms and pans in 2d with a camera.     Silverlight had a deep zoom,
short sounds, some of XNA...     A proper raster engine would have been
nicer.   one that allocates at most the amount of screen memory for the
display, not the source image. ...  but we can live with the mipmaps and
weird solutions .   If you open a giant blueprint in Autocad it could take
100 Meg to load... but it uses whatever your screen needs.. or less now
with RLE... , then... it has real time pan and zoom.  These image engines
might be open source, it was hi tech 15 years ago...  BTW.. now that MSFT
has XAMARIN we can expect Android support and MacOSX to go down.

**Comment by soernt** [comment]
+1 - That is really needed nowadays.

**Comment by InquisitorJax** [comment]
yeah... just throwing in my 2 cents here. UWP support would be really great - perhaps an eta or statement of intent?

**Comment by TomSoderling** [comment]
SharpDX 3 supports UWP now, so I simply recompiled this library to support UWP

**Comment by bukal260** [comment]
Could someone add this recompiled version as Nuget package? I need it badly...

**Comment by TomSoderling** [comment]
Sorry, I over-simplified the issue in my previous comment.  I'm working on a PR for this

**Comment by TomSoderling** [comment]
Just completed it.  I'd appreciate any feedback. https://github.com/praeclarum/NGraphics/pull/63

**Comment by philipogorman** [comment]
+1 for this we use xamairn forms for ios, droid and uwp,

**Comment by TomSoderling** [comment]
Good news! PR #63 was just merged.  Hopefully this will be released soon.

**Comment by InquisitorJax** [comment]
fantastic! Thanks @TomSoderling

**Comment by Abhijeet-Xamarin** [comment]
UWP + NGraphics 
"WinUniversal does not support saving to files. Please use the Stream override instead."
But I don't have stream its in canvas... any suggestions here.

**Comment by tarunagrawal1990** [comment]
Hi,

Is there any update on this.
I am trying to use NControl.Controls.NetCore in Xamarin.Forms project, but it is not working.

There is no compile time error but the controls are not rendering.

I have seen that I need to call initialization method for NControls i.e. NControls.Init(), but it seems that it is not available in UWP.

I have also tried with NControlViewRenderer.Init() but the result is same.

Any help would be appreciated.
Thanks.
